packages/flutter_locations/lib/src/ui/widgets/location_maps.dart

```diff
@@ -0,0 +1,379 @@
+import "dart:async";
+import "dart:io";
+
+import "package:dio/dio.dart";
+import "package:dio_cache_interceptor_hive_store/dio_cache_interceptor_hive_store.dart";
+import "package:flutter/foundation.dart";
+import "package:flutter/material.dart";
+import "package:flutter_hooks/flutter_hooks.dart";
+import "package:flutter_locations/src/config/locations_options.dart";
+import "package:flutter_locations/src/config/map_options.dart";
+import "package:flutter_locations/src/util/scope.dart";
+import "package:flutter_map/flutter_map.dart";
+import "package:flutter_map_cache/flutter_map_cache.dart";
+import "package:flutter_map_cancellable_tile_provider/flutter_map_cancellable_tile_provider.dart";
+import "package:hooks_riverpod/hooks_riverpod.dart";
+import "package:latlong2/latlong.dart";
+import "package:path_provider/path_provider.dart";
+import "package:platform_maps_flutter/platform_maps_flutter.dart"
+    as platform_maps;
+import "package:uuid/uuid.dart" as uuid;
+
+/// Entry widget receiving builder.
+class LocationsUserStory extends StatelessWidget {
+  ///
+  const LocationsUserStory({
+    required this.options,
+    super.key,
+  });
+
+  /// Options for the [LocationsUserStory].
+  final LocationsOptions options;
+
+  @override
+  Widget build(BuildContext context) {
+    var map = LocationsMap(
+      options: options.mapOptions,
+    );
+    return ProviderScope(
+      child: LocationsScope(
+        options: options,
+        child: options.builder(context, () {}, map),
+      ),
+    );
+  }
+}
+
+/// A function for determining the size of the cache.
+Future<int> calculateCacheSize(String path) async {
+  var directory = Directory(path);
+  var totalSize = 0;
+
+  try {
+    if (directory.existsSync()) {
+      var files = directory.listSync(recursive: true);
+      for (var file in files) {
+        if (file is File) {
+          totalSize += await file.length();
+        }
+      }
+    }
+  } on Exception catch (e) {
+    debugPrint("Error calculating cache size: $e");
+  }
+
+  return totalSize;
+}
+
+/// Provides path to retrieve cached data.
+final cachePathProvider = FutureProvider<String?>((ref) async {
+  if (kIsWeb) return null;
+  var cacheDirectory = await getTemporaryDirectory();
+  var cacheSizeBytes = await calculateCacheSize(cacheDirectory.path);
+
+  debugPrint("Cache size: ${cacheSizeBytes / 1024 / 1024} MB");
+  return cacheDirectory.path;
+});
+
+/// Retrieves cached data
+final cacheProvider = FutureProvider<HiveCacheStore>((ref) async {
```

Comment:
we should not use riverpod within a user-story, as not all apps that use this might want to depend on riverpod